app/domain/authentication/validate_origin.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Authentication
+  ValidateOrigin = CommandClass.new(
+    dependencies: {
+      get_role_by_login: GetRoleByLogin.new
+    },
+    inputs: %i(input)
+  ) do
+
+    def call
+      validate_origin
+    end
+
+    private
+
```

Comment:
I would put the two lines of code in `validate_origin` directly inside `call` (after all, the class name `ValidateOrigin` describes what the `call` method does, so it's already described by that).